processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -158,12 +160,36 @@ public GroupByQueryQueryToolChest(
       }
 
       final Sequence<Row> subqueryResult = mergeGroupByResults(subquery, runner, context);
-      final List<AggregatorFactory> aggs = Lists.newArrayList();
-      for (AggregatorFactory aggregatorFactory : query.getAggregatorSpecs()) {
-        aggs.addAll(aggregatorFactory.getRequiredColumns());
+
+      // check that all fieldName parameters in the outer query match up with a name parameter in the inner query
+      // for an aggregator or a post aggregator
+      Set<String> innerFieldNames = Sets.newHashSet();
+      for (AggregatorFactory innerAggregator : subquery.getAggregatorSpecs()) {
+        innerFieldNames.add(innerAggregator.getName());
+      }
+      for (PostAggregator innerPostAggregator : subquery.getPostAggregatorSpecs()) {
+        innerFieldNames.add(innerPostAggregator.getName());
+      }
+
+      for (AggregatorFactory outerAggregator : query.getAggregatorSpecs()) {
+        for (final String fieldName : outerAggregator.requiredFields()) {
+          if (!innerFieldNames.contains(fieldName)) {
+            throw new IllegalArgumentException(
+                String.format("Subquery must have an aggregator or post aggregator with name '%s'", fieldName)
+            );
+          }
+        }
       }
 
       // We need the inner incremental index to have all the columns required by the outer query
+      final List<AggregatorFactory> aggs = Lists.newArrayList(subquery.getAggregatorSpecs());
+      for (PostAggregator postAgg : subquery.getPostAggregatorSpecs()) {
+        // This causes the post aggregators from the inner query to be copied to the incremental index so that they are
+        // available as columns for the outer query. The data isn't modified by the aggregator since it has already
+        // been fully grouped by the inner query. Somewhat of a hack to get this working with an incremental index.
+        aggs.add(new DoubleSumAggregatorFactory(postAgg.getName(), postAgg.getName()));
```

Comment:
I think it would work to bring back the requiredColumns and use those instead of a DoubleSumAggregatorFactory. it's kind of silly since we really just want to retain the postAggregator field from the inner query (that's what the DoubleSum would do for numeric values) but there is not really a simple way to do that with an incremental index.

maybe an AggregatorFactory that just returns what it's given first?